lib/httpClient.ts

```diff
@@ -29,11 +31,21 @@ export default abstract class HTTPClient {
             retryCondition: (error: any) => {
                 return isNetworkError(error) || isRetryableError(error);
             },
-        })
+        });
+
+        // If we've hit the rate limit, don't make further requests before
+        // the reset time.
+        this.client.interceptors.request.use((config) => {
+            return this.checkLimit(config);
+        });
 
         this.client.interceptors.response.use(
             (response) => response,
             (error) => {
+                const limit = parseLimitFromResponse(error.response);
+                const limitKey = `${limit.scope}:${limit.type}`;
```

Comment:
What does the key look like in case `limit.scope` is not populated? Remember, the `scope` is only valid for `api` routes (`api` meaning not `ingest` or `query`).

Could this cause problems?

Reply:
hmm I am not sure what should it be, now its `unknown:ingest/query`, what should it be?

Reply:
> hmm I am not sure what should it be, now its `ingest/query:unknown`, what should it be?

Is it really `ingest/query:unknown`? `${limit.scope}:${limit.type}` looks like it will resolve to `unknown:ingest/query`. Either way, this seems fine. Just wanted to double check.

Reply:
yeah I realized earlier that the order was reversed. So, I followed same style in Go, and created a function to handle the key.